shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -2647,34 +2568,16 @@ public void release() {
      * WireCommand.
      */
     @Data
-    public static final class ConditionalAttributeUpdate {
-        public static final byte REPLACE = (byte) 1; // AttributeUpdate of type AttributeUpdateType.Replace.
-        public static final byte REPLACE_IF_EQUALS = (byte) 4; // AttributeUpdate of type AttributeUpdateType.ReplaceIfEquals.
-        public static final int LENGTH = 4 * Long.BYTES + 1; // UUID (2 longs) + oldValue + newValue + updateType (1 byte)
-
-        private final UUID attributeId;
-        private final byte attributeUpdateType;
-        private final long newValue;
-        private final long oldValue;
-
-        public void writeFields(DataOutput out) throws IOException {
-            out.writeLong(attributeId.getMostSignificantBits());
-            out.writeLong(attributeId.getLeastSignificantBits());
-            out.writeByte(attributeUpdateType);
-            out.writeLong(newValue);
-            out.writeLong(oldValue);
-        }
-
-        public static ConditionalAttributeUpdate readFrom(DataInput in, int length) throws IOException {
-            UUID attributeId = new UUID(in.readLong(), in.readLong());
-            byte attributeUpdateType = in.readByte();
-            long newValue = in.readLong();
-            long oldValue = in.readLong();
-            return new ConditionalAttributeUpdate(attributeId, attributeUpdateType, newValue, oldValue);
-        }
+    public static final class BatchInfo {
+        private final long batchId;
+        private final int seqNo;
+        private final boolean isLastSegment;
 
-        public int size() {
-            return LENGTH;
+        public static BatchInfo readFrom(DataInput in, int length) throws IOException {
```

Comment:
Please put the serialization code next to the read code. Currently your serialization code is in the `MergeSegmentCommand` . I suggest you bring that code here.

Reply:
fixed.